nomad/local/grapl-local-infra.nomad

```diff
@@ -142,4 +167,95 @@ job "grapl-local-infra" {
       }
     }
   }
+
+  group "kafka" {
+    network {
+      mode = "bridge"
+    }
+
+    task "kafka" {
+      driver = "docker"
+
+      config {
+        image = "confluentinc/cp-kafka:6.2.0"
+        # Hitting OOM errors
+        memory_hard_limit = 2048
+      }
+
+      env {
+        KAFKA_BROKER_ID = 1
+        KAFKA_ZOOKEEPER_CONNECT = "localhost:${var.ZOOKEEPER_PORT}"
+        KAFKA_LISTENER_SECURITY_PROTOCOL_MAP = "PLAINTEXT:PLAINTEXT"
+        KAFKA_LISTENERS = "PLAINTEXT://localhost:${var.KAFKA_BROKER_PORT}"
+        KAFKA_ADVERTISED_LISTENERS = "PLAINTEXT://localhost:${var.KAFKA_BROKER_PORT}"
+        KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR = 1
+        KAFKA_TRANSACTION_STATE_LOG_MIN_ISR = 1
+        KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR = 1
+        KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS = 0
+        KAFKA_JMX_PORT = var.KAFKA_JMX_PORT
+        KAFKA_JMX_HOSTNAME = "localhost"
+        KAFKA_LOG4J_ROOT_LOGLEVEL = "INFO"
+      }
+    }
+
+      service {
+        name = "kafka-broker"
+        port = var.KAFKA_BROKER_PORT
+        tags = ["kafka"]
+
+        connect {
+          sidecar_service {
+            proxy {
+              # connect to zookeeper
+              upstreams {
+                destination_name = "zookeeper"
+                local_bind_port = var.ZOOKEEPER_PORT
+              }
+            }
+          }
+        }
+
+      }
+
+
+  }
+
+  group "zookeeper" {
+    network {
+      mode = "bridge"
+    }
+
+    task "zookeeper" {
+      driver = "docker"
+
+      config {
+        image = "confluentinc/cp-zookeeper:6.2.0"
+      }
+
+      env {
+        ZOOKEEPER_CLIENT_PORT = var.ZOOKEEPER_PORT
+        ZOOKEEPER_TICK_TIME = 2000
+      }
+    }
+
+      service {
+        name = "zookeeper"
+        port = var.ZOOKEEPER_PORT
+        tags = ["zookeeper"]
+
+        connect {
+          sidecar_service {
```

Comment:
Just for my clarification, did this not work if you didn't have the bidirectional consul connect setup? I don't see anything passed into zookeeper regarding kafka's location. Does zookeeper establish additional connections to kafka or does it just keep open the connection that kafka establishes?

I wonder if you could just reduce this down to:

```
service {
  name = "zookeeper"
  port = var.ZOOKEEPER_PORT
  tags = ["zookeeper"]
  
  connect {
    sidecar_service { }
  }
}
```

Reply:
That seems to work 👍